battle.go

```diff
@@ -1,5 +1,117 @@
 package pokemonbattlelib
 
+import "fmt"
+
 type Battle struct {
 	// TODO
+	State   BattleState
+	Player1 *Client
+	Player2 *Client
+}
+type BattleState int
+
+const (
+	BEFORE_START BattleState = iota
+	BATTLE_START
+	BEFORE_PLAYER1_TURN
+	AFTER_PLAYER1_TURN
+	BEFORE_PLAYER2_TURN
+	AFTER_PLAYER2_TURN
+	BATTLE_END
+)
+
+// Creates a new battle instance that clients can connect to
+func NewBattle() *Battle {
+	return &Battle{State: BEFORE_START}
+}
+
+// Updates the battle state and dispatches calls to custom event hooks
+func (b *Battle) Dispatch(state BattleState) {
+	b.State = state
+	if handler, ok := b.Player1.EventHooks[state]; ok {
+		handler()
+	}
+	if handler, ok := b.Player2.EventHooks[state]; ok {
+		handler()
+	}
+}
+
+// Adds a player to the battle
+// Fails if two players are already connected
+func (b *Battle) AddClient(c *Client) error {
+	if b.Player1 == nil {
+		b.Player1 = c
+	} else if b.Player2 == nil {
+		b.Player2 = c
+	} else {
+		return fmt.Errorf("battle already has two clients connected")
+	}
+	return nil
+}
+
+// Removes a player by their ID from the battle
+func (b *Battle) RemoveClient(clientID int64) error {
+	if b.Player1 != nil && b.Player1.ID == clientID {
+		b.Player1 = nil
+	} else if b.Player2 != nil && b.Player2.ID == clientID {
+		b.Player2 = nil
+	} else {
+		return fmt.Errorf("no client with ID '%v' found", clientID)
+	}
+	return nil
+}
+
+// Swaps player 1 with player 2 in a battle
+func (b *Battle) SwapPlayers() error {
+	if b.Player1 == nil || b.Player2 == nil {
+		return fmt.Errorf("battle does not have two clients/players")
+	}
+	b.Player1, b.Player2 = b.Player2, b.Player1
+	return nil
+}
```

Comment:
Is this function necessary?

Reply:
Only thought this was needed in case the player wanted to reverse the first opponent to act and not deal with the battle fields manually. I guess if it becomes an array then i'm not sure?